perception/image_projection_based_fusion/src/utils/utils.cpp

```diff
@@ -13,8 +13,30 @@
 // limitations under the License.
 
 #include "image_projection_based_fusion/utils/utils.hpp"
+
 namespace image_projection_based_fusion
 {
+Eigen::Vector2d calcRawImageProjectedPoint(
+  const image_geometry::PinholeCameraModel & pinhole_camera_model, const cv::Point3d & point3d)
+{
+  cv::Point2d rectified_image_point = pinhole_camera_model.project3dToPixel(point3d);
+
+  cv::Point2d raw_image_point = rectified_image_point;
+
+  try {
+    const bool need_unrectify = std::any_of(
+      pinhole_camera_model.cameraInfo().d.begin(), pinhole_camera_model.cameraInfo().d.end(),
+      [](double distortion) { return distortion != 0.0; });
+
+    if (need_unrectify) {
+      raw_image_point = pinhole_camera_model.unrectifyPoint(rectified_image_point);
+    }
+  } catch (cv::Exception & e) {
+    RCLCPP_WARN_STREAM(rclcpp::get_logger("image_projection_based_fusion"), e.what());
+  }
```

Comment:
```suggestion
  const bool need_unrectify = std::any_of(
    pinhole_camera_model.cameraInfo().d.begin(), pinhole_camera_model.cameraInfo().d.end(),
    [](double distortion) { return distortion != 0.0; });

  if (need_unrectify) {
    raw_image_point = pinhole_camera_model.unrectifyPoint(rectified_image_point);
  }
```
Regarding https://github.com/autowarefoundation/autoware.universe/pull/7030/files#r1604255216, here's my proposal.